src/infill.cpp

```diff
@@ -82,59 +82,55 @@ void Infill::generate(std::vector<VariableWidthLines>& toolpaths,
-        constexpr coord_t gap_wall_count = 1; // Only need one wall here, less even, in a sense.
-        constexpr coord_t wall_0_inset = 0; // Don't apply any outer wall inset for these. That's just for the outer wall.
-        WallToolPaths wall_toolpaths(inner_contour, infill_line_width, gap_wall_count, wall_0_inset, settings);
-        std::vector<VariableWidthLines> gap_fill_paths = wall_toolpaths.getToolPaths();
-
-        // Add the gap filling to the toolpaths and make the new inner contour 'aware' of the gap infill:
-        // (Can't use getContours here, because only _some_ of the lines Arachne has generated are needed.)
-        Polygons gap_filled_areas;
-        for (const auto& var_width_line : gap_fill_paths)
+    // It does not make sense to print a pattern in a small region. So the infill region
+    // is split into a small region that will be filled with walls and the normal region
+    // that will be filled with the pattern. This split of regions is not needed if the
+    // infill pattern is concentric or if the small_area_width is zero.
+    if (pattern != EFillMethod::CONCENTRIC && small_area_width > 0)
+    {
+        // Split the infill region in a narrow region and the normal region.
+        Polygons small_infill = inner_contour;
+        inner_contour = inner_contour.offset(-small_area_width / 2).offset(small_area_width / 2);
+        small_infill = small_infill.intersection(small_infill.difference(inner_contour));
+        small_infill = Simplify(max_resolution, max_deviation, 0).polygon(small_infill);
+
+        // Small corners of a bigger area should not be considered narrow and are therefore added to the bigger area again.
+        for (PolygonsPart& small_infill_part : small_infill.splitIntoParts())
         {
-            VariableWidthLines thin_walls_only;
-            for (const auto& extrusion : var_width_line)
+            if (small_infill_part.offset(-infill_line_width / 2).offset(infill_line_width / 2).area() < infill_line_width * infill_line_width * 10)
             {
-                if (extrusion.is_odd && extrusion.inset_idx == 0)
+                if (! inner_contour.intersection(small_infill_part.offset(infill_line_width / 4)).empty())
                 {
-                    Polygon path;
-                    for (const auto& junction : extrusion.junctions)
-                    {
-                        path.add(junction.p);
-                    }
-                    if (path.polygonLength() >= infill_line_width * 4) // Don't fill gaps that are very small (with paths less than 2 line widths long, 4 back and forth).
-                    {
-                        gap_filled_areas.add(path);
-                        if (fill_gaps)
-                        {
-                            thin_walls_only.push_back(extrusion);
-                        }
-                    }
+                    small_infill = small_infill.difference(small_infill_part);
```

Comment:
I think you can be done smarter by omitting the difference operation as this is quite an expensive operation. I assume this is done so the `small_infill` contains all regions that are not printed yet. As you only print a subset of parts in this loop it is possible that some parts are not yet printed. Is this correct?

But if that is the case, why not do something like:
```cpp
auto small_infill_parts = small_infill.splitIntoParts();
small_infill.clear();
for (PolygonsPart& small_infill_part : small_infill_parts)
{
    if (condition_to_add_part)
    {
         // do stuff
    }
    else
    {
         small_infill.add(small_infill_part);
    }
}
```
Benefit of this is that no expensive difference is executed, and there is no chance of introducing small polygon-shilfers due to boolean operation artifacts.